demos/object_detection_demo_ssd_async/detection_model.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+// Copyright (C) 2018-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+#include "detection_model.h"
+#include <samples/args_helper.hpp>
+#include <samples/slog.hpp>
+
+using namespace InferenceEngine;
+
+DetectionModel::DetectionModel(std::string modelFileName, float confidenceThreshold, bool useAutoResize, const std::vector<std::string>& labels)
```

Comment:
All three `DetectionModel`s functions almost don't depend on `DetectionModel`s state. This allows to define them as separate functions. This will simplify class hierarchy by removing one layer.

Reply:
logically they related to DetectionModel, so they should be kept inside class.

Reply:
The point is that the whole class is not needed. You can put the functions in a separate file, but there is no need in additional lvl of abstraction.

Reply:
preprocess function is common for all inherited classes, so we should keep it as is.

Reply:
Ok.